farmhelper-mod/src/main/java/com/jelly/farmhelper/macros/SShapeCropMacro.java

```diff
@@ -290,33 +290,35 @@ public void onTick() {
             case RIGHT:
                 if (!waitForChangeDirection.isScheduled()) {
                     LogUtils.debugLog("Middle of row, going right");
-                    updateKeys(FarmConfig.cropType != MacroEnum.CACTUS && FarmConfig.cropType != MacroEnum.PUMPKIN_MELON && shouldWalkForwards(), FarmConfig.cropType == MacroEnum.CACTUS && shouldPushBack(), true, false, true);
+                    updateKeys((FarmHelper.config.VerticalMacroType != SMacroEnum.PUMPKIN_MELON.ordinal()) &&
+                                (FarmHelper.config.SShapeMacroType != SMacroEnum.CACTUS.ordinal()) && shouldWalkForwards(), (FarmHelper.config.SShapeMacroType == SMacroEnum.CACTUS.ordinal()) && shouldPushBack(), true, false, true);
                 }
                 return;
             case LEFT:
                 if (!waitForChangeDirection.isScheduled()) {
                     LogUtils.debugLog("Middle of row, going left");
-                    updateKeys(FarmConfig.cropType != MacroEnum.CACTUS && FarmConfig.cropType != MacroEnum.PUMPKIN_MELON && shouldWalkForwards(), FarmConfig.cropType == MacroEnum.CACTUS && shouldPushBack(), false, true, true);
+                    updateKeys((FarmHelper.config.VerticalMacroType != SMacroEnum.PUMPKIN_MELON.ordinal()) &&
+                        (FarmHelper.config.SShapeMacroType != SMacroEnum.CACTUS.ordinal()) && shouldWalkForwards(), (FarmHelper.config.SShapeMacroType == SMacroEnum.CACTUS.ordinal()) && shouldPushBack(), false, true, true);
                 }
                 return;
             case SWITCH_START:
                 if (!waitForChangeDirection.isScheduled()) {
                     if (switchBackwardsDirection) {
                         LogUtils.debugFullLog("Continue backwards");
-                        updateKeys(false, true, prevState == State.RIGHT, prevState == State.LEFT, true);
+                        updateKeys(false, true, prevState == State.RIGHT, prevState == State.LEFT, false);
```

Comment:
Change the attack to `true` to make it more "natural", most of the ppl still holds the attack button while changing the lanr

Reply:
@CatalizCS Check this also, ppl will complain (again) about this

Reply:
added toggle in latest commit